include/ensmallen_bits/adam/adam.hpp

```diff
@@ -99,6 +103,15 @@ class AdamType
            const bool shuffle = true,
            const bool resetPolicy = true);
 
+  AdamType(const UpdateRule& updateRule,
```

Comment:
Instead of introducing a new constructor, we could also put ``updateRule`` and ``decayPolicy`` at the end of the first one and use the default value, let me know what you think.

Reply:
I dont think that will work , if you look at how the constructor works,
[https://github.com/mlpack/ensmallen/blob/a8f3022afba52bfc22d6423feba0492b29a198d5/include/ensmallen_bits/adam/adam_impl.hpp#L23-L42](url)
We explicitly pass parameters( epsilon , beta1 and beta2 ) into the Update Rule, if we where to replace it , it would pass those values to the Update , which is wrong because we are passing our own Update and Decay Policies, and if we put the policies in the beginning , then we to pass them to access other paramters and the default parameter values are still passed .
If there is something that I have missed then please correct me .

Reply:
You are right, so perhaps it's not the best idea to provide a constructor to set the update rule, just to keep the interface simple. I could also use ``optimizer.UpdatePolicy() = policy;`` to get the same functionality. So I would remove the extra constructor, if you are fine with that, I think I suggested this in the first place.